src/osip-core/src/test/java/edu/kit/pse/osip/core/io/files/ExtendedParserTest.java

```diff
@@ -0,0 +1,184 @@
+package edu.kit.pse.osip.core.io.files;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+/**
+ * Tests the extended parser class
+ *
+ * @author Hans-Peter Lehmann
+ * @version 1.0
+ */
+public class ExtendedParserTest {
+    /**
+     * Tests numbers can be read
+     * @throws ParserException If something goes wrong
+     */
+    @Test
+    public void testNumber() throws ParserException {
+        ExtendedParser parser = new ExtendedParser("0");
+        assertEquals(0f, parser.readNumber(), 0.0001);
+        
+        parser = new ExtendedParser("424242");
+        assertEquals(424242f, parser.readNumber(), 0.0001);
+        
+        parser = new ExtendedParser("0.05");
+        assertEquals(0.05f, parser.readNumber(), 0.0001);
+
+        parser = new ExtendedParser("45.67");
+        assertEquals(45.67f, parser.readNumber(), 0.0001);
+    }
+    
+    /**
+     * Tests factors can be read
+     * @throws ParserException If something goes wrong
+     */
+    @Test
+    public void testFactor() throws ParserException {
+        ExtendedParser parser = new ExtendedParser("-5");
+        assertEquals(-5f, parser.readFactor(), 0.0001);
+        
+        parser = new ExtendedParser("--3");
+        assertEquals(3f, parser.readFactor(), 0.0001);
+        
+        parser = new ExtendedParser("10");
+        assertEquals(10f, parser.readFactor(), 0.0001);
+        
+        parser = new ExtendedParser("varname");
+        parser.variables.put("varname", 42f);
+        assertEquals(42f, parser.readFactor(), 0.0001);
+
+        parser = new ExtendedParser("-varname");
+        parser.variables.put("varname", 42f);
+        assertEquals(-42f, parser.readFactor(), 0.0001);
+    }
+    
+    /**
+     * Test invalid numbers
+     */
+    @Test
+    public void testInvalidNumbers() {        
```

Comment:
readNumber liest nur so weit, bis keine Zahlen mehr vorkommen. Also zum Beispiel bis zu einer schließenden Klammer oder bis zu diesem a. Dass `a` danach in diesem Kontext ein unerwartetes Zeichen ist,  weiß es nicht.